src/tiled/command.cpp

```diff
@@ -30,39 +30,70 @@
+
+    QString finalExecutable = replaceVariables(executable);
+    QFileInfo mFile(finalExecutable);   //Check if executable is a path or not
+
+    if (mFile.exists() && mFile.isFile()) {
+        finalWorkingDirectory.replace(QLatin1String("%executablepath"),
+                            QString(QLatin1String("%1")).arg(mFile.absolutePath()));
+    } else {
+        finalWorkingDirectory.replace(QLatin1String("%executablepath"),
+                            QString(QLatin1String("")));
+    }
+
+    return finalWorkingDirectory;
+}
+
 QString Command::finalCommand() const
 {
-    QString finalCommand = command;
+    QString finalCommand = QString(QLatin1String("%1 %2")).arg(executable).arg(arguments);
+
+    return replaceVariables(finalCommand);
+}
+
+QString Command::replaceVariables(const QString &string, bool quotePaths) const
+{
+    QString finalString = string;
 
     // Perform variable replacement
     if (Document *document = DocumentManager::instance()->currentDocument()) {
         const QString fileName = document->fileName();
 
-        finalCommand.replace(QLatin1String("%mapfile"),
-                             QString(QLatin1String("\"%1\"")).arg(fileName));
+        finalString.replace(QLatin1String("%mapfile"),
+                            QString(QLatin1String("\"%1\"")).arg(fileName));
 
         QFileInfo fileInfo(fileName);
         QString mapPath = fileInfo.absolutePath();
-        finalCommand.replace(
+
+        finalString.replace(
             QLatin1String("%mappath"),
             QString(QLatin1String("\"%1\"")).arg(mapPath));
 
         if (MapDocument *mapDocument = qobject_cast<MapDocument*>(document)) {
             if (const Layer *layer = mapDocument->currentLayer()) {
-                finalCommand.replace(QLatin1String("%layername"),
-                                     QString(QLatin1String("\"%1\"")).arg(layer->name()));
+                finalString.replace(QLatin1String("%layername"),
+                                    QString(QLatin1String("\"%1\"")).arg(layer->name()));
             }
         }
 
         if (MapObject *currentObject = dynamic_cast<MapObject *>(document->currentObject())) {
-            finalCommand.replace(QLatin1String("%objecttype"),
-                                 QString(QLatin1String("\"%1\"")).arg(currentObject->type()));
-            finalCommand.replace(QLatin1String("%objectid"),
-                                 QString(QLatin1String("\"%1\"")).arg(currentObject->id()));
+            finalString.replace(QLatin1String("%objecttype"),
+                                QString(QLatin1String("\"%1\"")).arg(currentObject->type()));
+            finalString.replace(QLatin1String("%objectid"),
+                                QString(QLatin1String("\"%1\"")).arg(currentObject->id()));
         }
     }
 
-    return finalCommand;
+    if (!quotePaths)
+        finalString.replace(QLatin1String("\""), QString(QLatin1String("")));
```

Comment:
I thought the same but couldn't figure out a case where a variable would contain quote. I'll do this just to be safe, though.